mhs/common/mhs_common/errors/ebxml_handler.py

```diff
@@ -0,0 +1,39 @@
+from typing import Tuple, AnyStr, Dict, Optional
+from xml.etree import ElementTree as ET
+
+from utilities.integration_adaptors_logger import IntegrationAdaptorsLogger
+
+from mhs_common.errors import ERROR_RESPONSE_DEFAULTS
+from mhs_common.messages.ebxml_error_envelope import EbxmlErrorEnvelope
+
+logger = IntegrationAdaptorsLogger('EBXML_ERROR_HANDLER')
+
+
+def handle_ebxml_error(code: int, headers: Dict, body: AnyStr) -> Tuple[int, Optional[AnyStr]]:
+    if code != 200 or body is None or body == '':
+        logger.warning('0001', 'Not HTTP 200 response. {Code} {Body}', {'Code': code, 'Body': body})
+        return code, body
+
+    if 'Content-Type' not in headers:
+        raise ValueError('No Content-Type header in Spine response, response cannot be handled!')
+
+    if headers['Content-Type'] != 'text/xml':
+        raise ValueError('Unexpected Content-Type {}!'.format(headers['Content-Type']))
+
+    parsed_body = ET.fromstring(body)
+
+    if not EbxmlErrorEnvelope.is_ebxml_error(parsed_body):
+        logger.info('0002', 'Not ebXML error.')
+        return code, body
+
+    ebxml_error_envelope: EbxmlErrorEnvelope = EbxmlErrorEnvelope.from_string(headers, body)
+
+    errors_text = ''
+    for idx, error_fields in enumerate(ebxml_error_envelope.errors):
+        all_fields = {**error_fields, **ERROR_RESPONSE_DEFAULTS}
```

Comment:
There is acceptance criteria:
`AND the errorType will be ebxml_error`
which is common for both - SOAP Faults and ebXML Errors

Reply:
Looking at [RT-109](https://gpitbjss.atlassian.net/browse/RT-109), it says:
> AND the errorType will be soap_fault

And looking at [RT-110](https://gpitbjss.atlassian.net/browse/RT-110), it says:
> AND the errorType will be ebxml_error

This might have been different when you were working on RT-109 though.

Reply:
Somebody has changed this. But OK, now there are two separate constants: one for SOAP faults and one for ebXML error.